### PR TITLE
@SimonSchick feat(node): v16.3

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V16.md#16.3.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/node/fs.d.ts [hunk 1]
```diff
@@ -1563,7 +1569,7 @@ declare module 'fs' {
      * If a flag is not provided, it defaults to `'r'`.
      */
     export function readFile(
-        path: PathLike | number,
+        path: PathOrFd,
```
Comment (peterblazejewicz) [inline types/node/fs.d.ts:1572]: can this be `PathOrFileDescriptor`?

File: types/node/index.d.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for non-npm package Node.js 16.0
+// Type definitions for non-npm package Node.js 16.1
```
Comment (peterblazejewicz) [inline types/node/index.d.ts:1]:
```suggestion
// Type definitions for non-npm package Node.js 16.3
```

**Review by peterblazejewicz — APPROVED** [review]

**Comment by SimonSchick** [comment]
Ready to merge

**Comment by SimonSchick** [comment]
Ready to merge

**Comment by Vampire** [comment]
@SimonSchick in https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/64140 I was sent here.
I think the name of `FlagAndOpenMode` is misleading.
It has two properties, `mode` of type `Mode` and `flag` of type `OpenMode`.
So the name parts in `FlagAndOpenMode` both just refer to the second property.
Maybe it should be `ModeAndOpenMode` or `FileModeAndOpenMode` or something similar.

**Pull request merged**